includes/models/model.llms.student.php

```diff
@@ -1841,4 +1909,171 @@ private function update_completion_status( $status, $object_id, $object_type, $t
+	 * @param  string $object_type  object type [lesson|section|course|track].
+	 * @return boolean
+	 */
+	public function mark_favorite( $object_id, $object_type ) {
+
+		// Short circuit if it's already favorited.
+		if ( $this->is_favorite( $object_id, $object_type ) ) {
+			return true;
+		}
+
+		return $this->update_favorite_status( 'favorite', $object_id, $object_type );
+
+	}
+
+	/**
+	 * Mark a lesson, section, course, or track unfavorite for the given user.
+	 *
+	 * @since    [version]
+	 *
+	 * @see    llms_mark_unfavorite() calls this function without having to instantiate the LLMS_Student class first.
+	 *
+	 * @param  int    $object_id    WP Post ID of the lesson, section, course, or track.
+	 * @param  string $object_type  object type [lesson|section|course|track].
+	 * @return boolean
+	 */
+	public function mark_unfavorite( $object_id, $object_type ) {
+
+		// Short circuit if it's not favorited.
+		if ( ! $this->is_favorite( $object_id, $object_type ) ) {
+			return true;
+		}
+
+		return $this->update_favorite_status( 'unfavorite', $object_id, $object_type );
+
+	}
+
+	/**
+	 * Triggers actions for favorite/unfavorite.
+	 *
+	 * Update the favorite status of a track, course, section, or lesson for the current student
+	 * Inserts / updates necessary user postmeta data.
+	 *
+	 * @since [version] Use filterable functions to determine if the object can be marked favorite.
+	 *
+	 * @param string $status      New status to update to, either "favorite" or "unfavorite".
+	 * @param int    $object_id   WP_Post ID of the object.
+	 * @param string $object_type The type of object. A lesson, section, course, or course_track.
+	 * @param string $trigger     String describing the reason for the status change.
+	 * @return boolean
+	 */
+	private function update_favorite_status( $status, $object_id, $object_type, $trigger = 'unspecified' ) {
+
+		$student_id = $this->get_id();
+
+		/**
+		 * Fires before a student's object favorite status is updated.
+		 *
+		 * The dynamic portion of this hook, `$status`, refers to the new completion status of the object,
+		 * either "favorite" or "unfavorite".
+		 *
+		 * @since [version]
+		 *
+		 * @param int    $student_id  WP_User ID of the student.
+		 * @param int    $object_id   WP_Post ID of the object.
+		 * @param string $object_type The type of object. A lesson, section, course, or course_track.
+		 * @param string $trigger     String describing the reason for the status change.
+		 */
+		do_action( "before_llms_mark_{$status}", $student_id, $object_id, $object_type, $trigger );
+
+		// Retrieve an instance of the object we're acting on.
+		if ( in_array( $object_type, llms_get_completable_post_types(), true ) ) {
+			$object = llms_get_post( $object_id );
+		} elseif ( in_array( $object_type, llms_get_completable_taxonomies(), true ) ) {
+			$object = get_term( $object_id, $object_type );
+		} else {
+			return false;
+		}
```

Comment:
why are we referring to completable post types or taxonomies? we only want this to be used on lessons for now. Maybe in the future for courses too, so either you check against a the lesson post type, or you create some kind of `llms_get_post_types_that_can_be_favorited()` that for the moment return an array of only one item:  `lesson`.

Reply:
There is something wrong, I didn't notice this code while coding, my fault. IDK, maybe Copilot :p